cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -299,7 +299,17 @@ predicate isImmutableOrUnobservable(Node n) {
 }
 
 /** Holds if `n` should be hidden from path explanations. */
-predicate nodeIsHidden(Node n) { n instanceof OperandNode and not n instanceof ArgumentNode }
+predicate nodeIsHidden(Node n) {
+  n instanceof OperandNode and not n instanceof ArgumentNode
+  or
+  StoreNodeFlow::flowThrough(n, _) and
```

Comment:
While trying to understand this change, I noticed the doc string for `flowThrough` is exactly the same as the one for `flowInto`, both for `StoreNodeFlow` and `ReadNodeFlow` (apart from a small type difference). Should the doc strings distinguish the two better?

Reply:
The QLDoc should probably be expanded a bit, yes. Both comments are technically accurate, but neither is very helpful :)

The `flowThrough` predicates are intended to be `private` and so the non-descriptive comment shouldn't affect users. But with that said, it's probably worth going over some of the predicates and write a more in-depth comment for ourselves.

Reply:
6c7a01d3d5b43d13a4ecbfb736a44877c57750fc adds some more helpful QLDoc. They're still very much similar, but that's because the predicates basically serve the same job (but one is for `ReadNode`s and the other is from `StoreNode`s).